app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -214,6 +214,13 @@ def save_hosts_inventory(ems, hashes, target = nil)
     end
   end
 
+  def save_hosts_storages_inventory(host, hashes, target = nil)
+    hashes.each do |mount|
+      storage_id = mount[:storage][:id]
+      HostsStorage.where(:host_id => host.id, :storage_id => storage_id).update_all(:read_only => mount[:read_only])
```

Comment:
This doesn't feel right in the save_inventory code.  In addition this will nail the database even if nothing has changed, which is not optimal.  I wonder if we can treat it like a child table like we do with the other save_inventory methods, so we can leverage save_inventory_multi (which does the optimal thing).  Something like:

```ruby
def save_hosts_storages_inventory(host, hashes, target = nil)
  target = host if target.nil?

  # Update the associated ids
  hashes.each do |h|
    h[:host_id]    = host.id # You may not need this, but you might for the unique key
    h[:storage_id] = h.fetch_path(:storage, :id)
  end
 
  host.hosts_storages(true)
  deletes =
    if target == host
      host.hosts_storages.dup
    else
      []
    end
 
  save_inventory_multi(:hosts_storages, host, hashes, deletes, [:host_id, :storage_id], nil, [:storage])
end
```